core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java

```diff
@@ -600,8 +601,13 @@ private void saveGatewayPillar(GatewayConfig gatewayConfig, Cluster cluster,
 
         ExposedService rangerService = exposedServiceCollector.getRangerService();
         List<String> rangerLocations = serviceLocations.get(rangerService.getServiceName());
+
         if (!CollectionUtils.isEmpty(rangerLocations)) {
-            serviceLocations.put(rangerService.getServiceName(), getSingleRangerFqdn(gatewayConfig.getHostname(), rangerLocations));
+            if (SdxClusterShape.LIGHT_DUTY.equals(cluster.getStack().getSdxClusterShape())) {
+                serviceLocations.put(rangerService.getServiceName(), getSingleRangerFqdn(gatewayConfig.getHostname(), rangerLocations));
+            } else if (SdxClusterShape.MEDIUM_DUTY_HA.equals(cluster.getStack().getSdxClusterShape())) {
```

Comment:
is it possible to have multiple gateway node for `LIGHT_DUTY`? I mean do we need this `if-else` at all? couldn't be it like all gateway node has ranger and `LIGHT_DUTY` has only one GW while `MEDIUM` has multiple and the result would be the same. In this case we won't need the cluster shape on CB side

Reply:
No, medium has multiple nodes called gateway, light duty has none.

Reply:
but in case of `LIGHT_DUTY` the master is `GATEWAY` type, while in case of `MEDIUM_DUTY_HA` not the master is the gateway. We could depend on that information and select those FQDNs. `Stack` has a method: `getGatewayInstanceMetadata` which could be used to get all the gateway instances.